File: 1850 D. Balanced Round/Solution.cpp
#include<bits/stdc++.h>
using namespace std; 

// Krishno_MD_KUET_CSE

void solve(){
    int n,k; cin>>n>>k;
    vector<int> v(n);
    for(int i=0; i<n; i++){cin>>v[i];}
    sort(v.begin(),v.end());
    int count = 1,a = 1;
    for(int i=1; i<n; i++){
        if(v[i]-v[i-1] > k){count = 1;}
        else{count++;}
        a = max(a,count);
    }
    cout<<n-a<<endl;
}

int main(){
    int t;
    cin >> t;
    while(t--){
        solve();
    }
    return 0;
}

File: 1850 D. Balanced Round/Solution.py
# Krishno_MD_KUET_CSE

t = int(input())

for i in range(t):
    n,k = map(int,input().split())
    v = list(map(int,input().split()))
    v.sort()
    count,a = 1,1
    for i in range(1,n):
        if v[i]-v[i-1] >k:
            count = 1
        else:
            count += 1
        a = max(a,count)
    print(n-a)
